packages/puppeteer-core/src/bidi/Page.ts

```diff
@@ -774,6 +775,136 @@ export class BidiPage extends Page {
     });
   }
 
+  override async cookies(...urls: string[]): Promise<Cookie[]> {
+    const normalizedUrls = (urls.length ? urls : [this.url()]).map(url => {
+      return new URL(url);
+    });
+
+    const bidiCookies = await this.#connection.send('storage.getCookies', {
+      partition: {
+        type: 'context',
+        context: this.mainFrame()._id,
+      },
+    });
+    return bidiCookies.result.cookies
+      .map(c => {
+        return this.#bidiToPuppeteerCookie(c);
+      })
+      .filter(c => {
+        return normalizedUrls.some(u => {
+          return BidiPage.#testUrlMatchCookie(c, u);
+        });
+      });
+  }
+
+  /**
+   * Check domains match.
+   * According to cookies spec, this check should match subdomains as well, but CDP
+   * implementation does not do that, so this method matches only the exact domains, not
+   * what is written in the spec:
+   * https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.3
+   */
+  static #testUrlMatchCookieHostname(
```

Comment:
some methods are static and some not? Let's move all of them as module functions to Cookie.ts